Nos/Controller/RefreshController.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// Defines a common interface for refreshing.
+@MainActor protocol RefreshController: Observable {
+    /// Whether a refresh should begin or not.
+    var shouldRefresh: Bool { get }
```

Comment:
I've been having trouble with properties that start with `should` in my own PRs. They were common in obj-c, but are not as much in swift, even though I like to continue using should, when I get a comment in my PRs asking to rename a property that started with should I didn't find something to reply because swift naming conventions don't mention anything about should. What's your take here?

Reply:
I considered `isRefreshing` but that wasn't right, since we immediately set it to `false` in `updateUIView`:

```
if refreshController.shouldRefresh {
    refreshController.setShouldRefresh(false)
```

This would mean `isRefreshing` is `false` while we're in the process of refreshing. Not the right name, for sure.

I thought about just using `refresh` but that seems wrong for a `Bool`. I even used `beginRefreshing` which matches the name of a method on `UIRefreshControl`. I didn't like that -- it was too easy to confuse these two lines:

```
refreshController.beginRefreshing()
refreshControl.beginRefreshing()
```

I might be ok with the name `startRefresh` or `startRefreshing` but I'd have to think about that a bit.